pkg/scheduler/queue/tenant_queues_test.go

```diff
@@ -416,15 +412,42 @@ func getOrAdd(t *testing.T, qb *queueBroker, tenantID TenantID, maxQueriers int)
 	reAddedQueue, err := qb.getOrAddTenantQueue(tenantID, maxQueriers)
 	assert.Nil(t, err)
 	assert.Equal(t, addedQueue, reAddedQueue)
-	return addedQueue
+	return addedQueue.localQueue
+}
+
+func (qb *queueBroker) getOrAddTenantQueue(tenantID TenantID, maxQueriers int) (*TreeQueue, error) {
+	_, err := qb.tenantQuerierAssignments.getOrAddTenant(tenantID, maxQueriers)
+	if err != nil {
+		return nil, err
+	}
+
+	queuePath := QueuePath{string(tenantID)}
+	return qb.tenantQueuesTree.getOrAddNode(queuePath)
+}
+
+func (qb *queueBroker) getQueue(tenantID TenantID) *TreeQueue {
+	tenant, err := qb.tenantQuerierAssignments.getTenant(tenantID)
+	if tenant == nil || err != nil {
+		return nil
+	}
+
+	queuePath := QueuePath{string(tenantID)}
+	tenantQueue := qb.tenantQueuesTree.getNode(queuePath)
+	return tenantQueue
+}
+
+func (qb *queueBroker) removeTenantQueue(tenantID TenantID) bool {
+	qb.tenantQuerierAssignments.removeTenant(tenantID)
+	queuePath := QueuePath{string(tenantID)}
+	return qb.tenantQueuesTree.deleteNode(queuePath)
 }
```

Comment:
Might be worth adding a note here that these methods should only be used in tests (otherwise someone might be tempted to move them to `tenant_queues.go`).